### PR TITLE
Backend Dependency Bump for 52

### LINKED ISSUE #44330: Start of 51 Cycle Routine Clojure Dependency Bump
[Notion doc about dependency bumping](https://www.notion.so/metabase/Dependency-Bumping-8ab452bbfc1f4131addd1653235c1b0f?pvs=4)

Subtasks:

- [x] Update all minor and patch versions and test. Revert breakage.
- [x] Update minor versions in modules

### PR DESCRIPTION
**Pull request by adam-james-v**
Closes #44330

## Major Bumps

[bigml/histogram 4.1.4 -> 5.0.0](https://github.com/bigmlcom/histogram/commit/ada60dbdf305f1b458aa9968bfc4fb363b68a27e)
[com.fasterxml.woodstox/woodstox-core 6.6.1 -> 7.0.0](https://github.com/FasterXML/woodstox/blob/master/release-notes/VERSION)

## Minor Bumps

[buddy/buddy-core 1.11.423 -> 1.12.0-420](https://github.com/funcool/buddy-core/blob/master/CHANGES.md)
[buddy/buddy-sign 3.5.351 -> 3.6.1-359](https://github.com/funcool/buddy-sign/blob/master/CHANGES.md)
[cheshire/cheshire 5.12.0 -> 5.13.0](https://cljdoc.org/d/cheshire/cheshire/5.13.0/doc/changelog) (changelog appears to be out of date)
--- continue adding changelog links here
clj-http/clj-http                  3.12.3 -> 3.13.0
com.google.guava               33.1.0-jre -> 33.2.1-jre
commons-codec/commons-codec        1.16.1 -> 1.17.1
commons-io/commons-io              2.15.1 -> 2.16.1
commons-net/commons-net            3.10.0 -> 3.11.1
commons-validator/commons-validator 1.8.0 -> 1.9.0
instaparse/instaparse              1.4.12 -> 1.5.0
net.clojars.wkok/openai-clojure    0.16.0 -> 0.18.1
org.apache.commons/commons-lang3   3.14.0 -> 3.15.0
org.apache.poi/poi                  5.2.5 -> 5.3.0
org.apache.poi/poi-ooxml            5.2.5 -> 5.3.0
org.apache.sshd/sshd-core          2.12.1 -> 2.13.1
org.jsoup/jsoup                    1.17.2 -> 1.18.1
org.threeten/threeten-extra         1.7.2 -> 1.8.0

### CODE REVIEW

**Comment by piranha** [comment]
Vertica errors seem legit, I propose removing it from this batch update.

**Review by camsaul — APPROVED** [review]
amazing thanks for this

**Comment by piranha** [comment]
Let's wait until 51 is cut to merge it :)

File: enterprise/backend/src/metabase_enterprise/sso/integrations/saml.clj
Comment (qnkhuat) [inline enterprise/backend/src/metabase_enterprise/sso/integrations/saml.clj:239]: is this legit? looks weird

Comment (piranha) [reply enterprise/backend/src/metabase_enterprise/sso/integrations/saml.clj:239]: yeah new core.xml started producing namespaces >_>

**Pull request merged**
